tmt/steps/__init__.py

```diff
@@ -14,6 +14,8 @@
 import click
 from click import echo
 
+import tmt
+import tmt.base
```

Comment:
I believe these two imports might be causing your issues.

```
__________________ ERROR collecting tests/unit/test_adjust.py __________________
tests/unit/test_adjust.py:3: in <module>
    from tmt.convert import relevancy_to_adjust
tmt/__init__.py:18: in <module>
    from tmt.base import Clean, Plan, Result, Run, Status, Story, Test, Tree
tmt/base.py:23: in <module>
    import tmt.steps
tmt/steps/__init__.py:73: in <module>
    class StepData(TypedDict, total=False):
tmt/steps/__init__.py:80: in StepData
    tests: Optional[List[tmt.base.Test]]
E   AttributeError: module 'tmt' has no attribute 'base'
```

`tmt` imports from `tmt.base`, which means `tmt.base` is read by the interpreter and processed. That leads to importing `tmt.steps` from `base.py`, then `steps.py` refers to `tmt.base.Test` - but `tmt.base` has not been fully imported yet, the process is still crunching `base.py` and `steps.py`, constructing the in-memory representation of those modules. This means `tmt` really has no `base` attribute, yet.

Try moving `import tmt.base` (is `import tmt` even needed?) under `TYPE_CHECKING` guard and a forward reference (`List['tmt.base.Test']`) to avoid referring to a partially imported module in runtime.

Reply:
seems resolved now, thanks @happz